airflow/executors/base_executor.py

```diff
@@ -356,13 +356,13 @@ def execute_async(
         """
         raise NotImplementedError()
 
-    def get_task_log(self, ti: TaskInstance) -> tuple[list[str], list[str]]:
+    def get_task_log(self, ti: TaskInstance, try_number: int) -> tuple[list[str], list[str]]:
         """
```

Comment:
wait this bit doesn't make sense to me.  if we already have the TI, why is it we also need to supply the try number?

Reply:
There is some explanation in the code for `read`: https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L376-L410

`try_number` is plumbed in from `read` to `_read` and then should have been sent to the executors, but it was a miss:
https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L274-L315